src/averageAges.js

```diff
@@ -37,7 +44,27 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = people.filter(person => person.sex === 'f');
+  let averageFemaleSum = 0;
+
+  if (withChildren) {
+    const femaleWithChild = filteredWomen.filter(person =>
+      people.some(child => child.mother === person.name));
+
+    averageFemaleSum = femaleWithChild.reduce((all, person) =>
+      all + (person.died - person.born), 0);
+
+    const averageAgeFemaleChild = averageFemaleSum / femaleWithChild.length;
+
+    return averageAgeFemaleChild;
+  }
+
+  averageFemaleSum = filteredWomen.reduce((all, person) =>
```

Comment:
You can optimize your code by calculating the 'averageFemaleSum' and 'averageAgeFemale' in one pass using the .reduce() method. This will improve the performance of your code.